library/src/animationManager.cxx

```diff
@@ -274,3 +275,75 @@ bool animationManager::LoadAtTime(double timeValue)
   return true;
 }
 }
+
+//----------------------------------------------------------------------------
+std::string animationManager::GetAnimationDescription()
+{
+  if (!this->Importer)
+    s
+    {
+      return "";
+    }
+
+  std::stringstream stream;
+  animationManager::AnimationInfo info;
+  if (this->Importer->GetInfoForAnimation(this->HasAnimation, this->ArrayIndexForAnimation, info))
+  {
+    stream << "Current Animation " << info.Name << ", "
+           << animationManager::ComponentToString(this->ComponentForAnimation) << "\n";
+  }
+  else
+  {
+    stream << "Not animating\n";
+  }
+  return stream.str();
+}
+//----------------------------------------------------------------------------
+void animationManager::CycleAnimationsForGLTF()
+{
+
+  assert(this->Importer);
+
+  int nIndex = this->Importer->GetNumberOfAnimations();
+  if (nIndex <= 0)
+  {
+    return;
+  }
+
+  if (this->HasAnimation)
+  {
+    this->ArrayIndexForAnimation = (this->ArrayIndexForAnimation + 1) % nIndex;
+  }
+  else
+  {
+    // Cycle through arrays looping back to -1
+    // -1 0 1 2 -1 0 1 2 ...
+    this->ArrayIndexForAnimation = (this->ArrayIndexForAnimation + 2) % (nIndex + 1) - 1;
+  }
+}
+//----------------------------------------------------------------------------
+void animationManager::CycleAnimations(CycleType type)
+{
+  if (!this->Importer)
+  {
+    return;
+  }
+
+  switch (type)
+  {
+    case (CycleType::NONE):
+      return;
+      break;
+    case (CycleType::GLTF):
+      this->CycleAnimationsForGLTF();
+      break;
+    case (CycleType::GLB):
+      this->CycleAnimationsForGLB();
+      break;
+    case (CycleType::FBX):
```

Comment:
I'm a bit confused as to why you would need to diferentiate between file format

Reply:
My lack of familiarity with how animation data is stored within these file types caused me to believe that it is a possibility that their data structures could differ. This led me to believe it would be necessary to differentiate since the array-based solution would not be universal.

Reply:
No worries! Im glad you are trying :)